cmd/livepeer/livepeer.go

```diff
@@ -296,6 +296,16 @@ func main() {
 			return
 		}
 		orch := core.NewOrchestrator(s.LivepeerNode)
+
+		// check whether or not the transcoder is available
+		go func() {
+			time.Sleep(1 * Second)
+			if !server.checkTranscoderAvailability(orch) {
```

Comment:
IIUC the logic in this func assumes the transcoder service has been started, and that happens in the line below `server.StartTranscodeServer`. If that's true, I am wondering why start this go routine before we start the transcoder service rather than after?

I don't think there's any guarantee this go routine won't start running before the transcoder service is live. 

I see the sleep statement and assuming it's helping mitigate this race condition, but I don't think it guarantees the execution order we're expecting as much as running this check after `StartTranscodeServer` has been executed.

wdyt?

Reply:
The `StartTranscoderServer` function blocks until the server exits, so we're at the mercy of the scheduler no matter where we spin off the goroutine. If 1 second looks to be too short, we can always bump it up.

We could disassemble `ListenAndServeTLS`, set up the listener manually, spin off the rest of the TLS server, and return from `StartTranscodeServer` with a channel to signal when the server has shut down.  That would give us some stronger scheduling assurances, but the amount of plumbing work isn't really justified for this minor usability check. We can always improve it later.

Reply:
Note that we can choose to put the goroutine around the StartTranscoder call as well. That doesn't really change the scheduling implications, but perhaps offers a "better feeling" when reading the code. For example,

```
go func() {
  go func() {
    server.StarTranscoder(...)
    tc <- struct{}{}
  }()
  time.Sleep(1 * time.Second)
  if !CheckTranscoderAvailability(orch) {
    tc <- struct{}{}
  }
}()
```

Reply:
Yes, definitely makes it more readable. I'll do that and increase the time.Sleep to 2 seconds